### ISSUE TITLE
Drop netstandard1.6 and netcoreapp1.1

### ISSUE THREAD

**Issue by TheAngryByrd** [TheAngryByrd/MiniScaffold#68]
## Description

With logary dropping support for netstandard1.6 https://github.com/haf/expecto/commit/6429c66eb50bc120f57aa87ef04bb293e36acb7d and talking with people in the community we should just drop netstandard1.6 and netcoreapp1.1.

## Error messages, screenshots

N/A

## Failing test, failing github repo, or reproduction steps

N/A

## Expected Behavior

N/A

## Known workarounds

N/A

## Other information

* Operating System:
    - [ ] windows [insert version here]
    - [ ] macOs [insert version]
    - [ ] linux [insert flavor/version here]
* Platform
    - [ ] dotnet core
    - [ ] dotnet full
    - [ ] mono
* Branch or release version: master

**Comment by rmunn** [comment]
The first time I tried to use MiniScaffold to start up a project, I had a little bit of trouble with getitng the dotnetcore1.1 SDK installed on my Linux dev machine, since MS no longer releases a Linux package for the dotnetcore1.1 SDK on current versions of Linux. My choices were to go back to an older version of Linux, or to install the dotnetcore 1.1 SDK by hand. I chose the latter, and it all worked.

I think my experience is another argument in favor of dropping netstandard1.6 and netcoreapp1.1 from the default template. If anyone still needs them, adding them is as simple as editing two .fsproj files and rerunning the build. So it's not a great burden to anyone who needs netcoreapp1.1, and it lowers the barrier to entry for people just getting started with .Net Core development.

**Comment by TheAngryByrd** [comment]
Lots of other popular libraries have dropped support for it like [Argu](https://www.nuget.org/packages/Argu), [Expecto](https://www.nuget.org/packages/Expecto/), [Giraffe](https://www.nuget.org/packages/Giraffe/1.2.0-preview-1), [Suave](https://www.nuget.org/packages/Suave/2.3.0-beta3), [Httpfs](https://www.nuget.org/packages/Http.fs/)

**Comment by rmunn** [comment]
The fact that three of those five projects you cite are (or were) maintained by Henrik Feldt shows just how prolific and active an F# coder he is. I'm seriously impressed by that guy. :smile:

But since most of them are (as I understand it) now actively maintained by other people, that means that those projects dropping dotnetcore1.1 *also* isn't just one man's opinion, so it can be taken as broadly representative. In short: I'm in favor of dropping it from the MiniScaffold also.

**Comment by Tarmil** [comment]
I agree. As another data point, when porting WebSharper to .NET Standard we went straight for netstandard2.0.

**Comment by wallymathieu** [comment]
I think just dropping `netcoreapp1.1` should not be a big issue. Why drop `netstandard1.6`?

**Comment by wallymathieu** [comment]
`Giraffe` depends on infrastructure that requires `netstandard2.0`.
`Argu`, `Expecto` and `Suave` are intended for running as executables (and less likely to actually care about other platforms than server side code) might  probably be less of a contentious issue. 
`Httpfs` never had support for anything lower than `netstandard2.0` from what I can see from nuget. So it might be that they need to use things not part of `netstandard1.6`.

**Comment by TheAngryByrd** [comment]
Is there a way to tell the project system to have netcoreapp2.0 test netstandard2.0 then netstandard1.6? I actually have no idea how which framework knows which library framework to choose when both have multitargeting enabled.

**Comment by wallymathieu** [comment]
Well, the easiest thing would be to start out with `netstandard1.6` and then bump it if you need to. That way you can run the tests on `netcoreapp2.0` without worrying. And if a library bumps, then it can be their decision to remove support for it.

If you really want to test a library built against `netstandard1.6` and `netstandard2.0` you would need some msbuild hacking to have it working. Perhaps a parameter to the build that disables `netstandard2.0` or overrides it:
```
<TargetFrameworks Condition=" '$(TestTargetBuildFramework)' == '' ">netstandard1.1;netstandard1.3;netstandard2.0</TargetFrameworks>
<TargetFrameworks Condition=" '$(TestTargetBuildFramework)' != '' ">$(TestTargetBuildFramework)</TargetFrameworks>
```

**Comment by TheAngryByrd** [comment]
More datapoints.  4.0 of [Npgsql](https://www.nuget.org/packages/Npgsql/4.0.0-preview1) has dropped netstandard1.3 and so did "netcore" of [Eventstore](https://www.nuget.org/packages/EventStore.ClientAPI.NetCore/4.1.0.23)

**Comment by wallymathieu** [comment]
We use `netstandard1.6` for most of our internal libraries. For some projects we use [EF](https://docs.microsoft.com/en-us/ef/core/) so we have to bump to `netstandard2.0`. Some internal libraries that are pure c# without bells and whistles we actually use `netstandard1.3` (but that's not so common).

**Comment by ScottHutchinson** [comment]
Version 0.8.0 seems to have a mismatch between src and tests projects:

src:
    <TargetFrameworks>netstandard2.0;netstandard1.6;net461</TargetFrameworks>

tests:
    <TargetFrameworks>netcoreapp2.0;netcoreapp1.1;net461</TargetFrameworks>

Shouldn't tests have netcoreapp1.6 instead of netcoreapp1.1 ?

**Comment by wallymathieu** [comment]
Has there been a release of netcoreapp1.6? From what I can see from the docs, netstandard1.6 is still compatible with 1.0: [dotnet standard versions](https://github.com/dotnet/standard/blob/master/docs/versions.md)

**Comment by ScottHutchinson** [comment]
@wallymathieu Looks like you're right. It appers that netcoreapp should always be either netcoreapp1.0 or netcoreapp2.0 then. But there will be a version 2.1: https://github.com/dotnet/core/blob/master/roadmap.md#technology-roadmaps

**Comment by wallymathieu** [comment]
looks like there is a [v1.1.6](https://github.com/dotnet/core/tree/v1.1.6)

**Comment by wallymathieu** [comment]
I think the paket guys know more about the weirdness with all of these framework tags. So perhaps someone from there could point in the right direction.

**Comment by TheAngryByrd** [comment]
We've already had to remove netcoreapp1.1 from the tests due to Expecto.  

And David Fowler is calling for people to move to netstandard2.0. 

https://twitter.com/davidfowl/status/1008453902058917888

**Comment by wallymathieu** [comment]
Sounds like reason enough to drop it.
